aptos-move/move-examples/metapixel/sources/catcha_token.move

```diff
@@ -0,0 +1,292 @@
+            mutable_uri,
+            mutable_token_description,
+            mutable_token_name,
+            mutable_token_properties,
+            mutable_token_uri,
+            tokens_burnable_by_creator,
+            tokens_freezable_by_creator,
+        };
+        move_to(&object_signer, catcha_collection);
+    }
+
+
+    inline fun collection_object(creator: &signer, name: &String): Object<CatchaCollection> {
+        let collection_addr = collection::create_collection_address(&signer::address_of(creator), name);
+        object::address_to_object<CatchaCollection>(collection_addr)
+    }
+
+
+    inline fun borrow_collection<T: key>(token: &Object<T>): &CatchaCollection {
+        let collection_address = object::object_address(token);
+        assert!(
+            exists<CatchaCollection>(collection_address),
+            error::not_found(ECOLLECTION_DOES_NOT_EXIST),
+        );
+        borrow_global<CatchaCollection>(collection_address)
+    }
+
+
+    fun mint_internal(
+        creator: &signer,
+        collection: String,
+        description: String,
+        name: String,
+        uri: String,
+        property_keys: vector<String>,
+        property_types: vector<String>,
+        property_values: vector<vector<u8>>,
+    ): ConstructorRef acquires CatchaCollection {
+        let constructor_ref = token::create(
+            creator,
+            collection,
+            description,
+            name,
+            option::none(),
+            uri,
+        );
+
+        let object_signer = object::generate_signer(&constructor_ref);
+
+        let collection_obj = collection_object(creator, &collection);
+        let collection = borrow_collection(&collection_obj);
+
+        let mutator_ref = if (
+            collection.mutable_token_description
+                || collection.mutable_token_name
+                || collection.mutable_token_uri
+        ) {
+            option::some(token::generate_mutator_ref(&constructor_ref))
+        } else {
+            option::none()
+        };
+
+        let burn_ref = if (collection.tokens_burnable_by_creator) {
+            option::some(token::generate_burn_ref(&constructor_ref))
+        } else {
+            option::none()
+        };
+
+        let catcha_token = CatchaToken {
+            burn_ref,
+            transfer_ref: option::none(),
+            mutator_ref,
+            property_mutator_ref: property_map::generate_mutator_ref(&constructor_ref),
+        };
+        move_to(&object_signer, catcha_token);
+
+        let properties = property_map::prepare_input(property_keys, property_types, property_values);
+        property_map::init(&constructor_ref, properties);
+
+        constructor_ref
+    }
+
+
+    public fun are_collection_tokens_freezable<T: key>(
+        collection: Object<T>,
+    ): bool acquires CatchaCollection {
+        borrow_collection(&collection).tokens_freezable_by_creator
+    }
+
+
+    public fun mint_catcha_token(
```

Comment:
This is a good start :) Can you talk to Lucas and Willing about how they're planning to mint? (e.g. users minting themselves, NPixel minting and airdropping to everyone, any delayed reveal, etc.)?